File: Frontend/Modules/Classes/article.js
export class Article
{
    constructor({
        id,
        title,
        description,
        link,
        author,
        pubDate,
        feedUrl,
        rssFeedId,
        urlToImage
    })
    {
        this.id = id;
        this.title = title;
        this.description = description;
        this.link = link;
        this.author = author;
        this.pubDate = pubDate;
        this.feedUrl = feedUrl;
        this.rssFeedId = rssFeedId;
        this.urlToImage = urlToImage;
    }
}

File: Backend/DomainModels/Article.cs
using System.Xml.Linq;

namespace DomainModels
{
    public class Article : BaseEntity
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public string Link { get; set; }
        public string? Author { get; set; }
        public DateTime? PubDate { get; set; }
        //public DateTime? PubDateParsed { get; set; } //new property...
        public string FeedUrl { get; set; }

        // Foreign key to RssFeed
        public int RssFeedId { get; set; }
        public RssFeed RssFeed { get; set; } // Navigation property to RssFeed

        public string UrlToImage { get; set; } // For image extraction
        public ICollection<Comment> Comments { get; set; }
        public ICollection<Feedback> Feedbacks { get; set; }
    }
}
